Build fails with new version of tt-metal

We have build errors with new version of tt-metal
https://github.com/tenstorrent/tt-mlir/actions/runs/11660244306/job/32462210096

```
/__w/tt-mlir/tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/dev_msgs.h:208:26: error: zero size arrays are an extension [-Werror,-Wzero-length-array]
  208 |     volatile uint8_t pad[8 - DebugNumUniqueRiscs];
      |                          ^~~~~~~~~~~~~~~~~~~~~~~
/__w/tt-mlir/tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/dev_msgs.h:222:27: error: zero size arrays are an extension [-Werror,-Wzero-length-array]
  222 |     volatile uint16_t pad[16 - DebugNumUniqueRiscs * 2];
```

--- a/runtime/include/tt/runtime/detail/ttmetal.h
+++ b/runtime/include/tt/runtime/detail/ttmetal.h
@@ -38,6 +38,7 @@
 #pragma clang diagnostic ignored "-Wunused-but-set-variable"
 #pragma clang diagnostic ignored "-Wlogical-op-parentheses"
 #pragma clang diagnostic ignored "-Wundefined-inline"
+#pragma clang diagnostic ignored "-Wzero-length-array"
 #define FMT_HEADER_ONLY
 #include "distributed/mesh_device.hpp"
 #include "impl/event/event.hpp"

--- a/runtime/include/tt/runtime/detail/ttnn.h
+++ b/runtime/include/tt/runtime/detail/ttnn.h
@@ -42,6 +42,7 @@
 #pragma clang diagnostic ignored "-Wc99-extensions"
 #pragma clang diagnostic ignored "-Wc++11-narrowing"
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma clang diagnostic ignored "-Wzero-length-array"
 
 #define FMT_HEADER_ONLY
 #include "distributed/mesh_device.hpp"

--- a/runtime/lib/common/system_desc.cpp
+++ b/runtime/lib/common/system_desc.cpp
@@ -28,6 +28,7 @@
 #pragma clang diagnostic ignored "-Wmismatched-tags"
 #pragma clang diagnostic ignored "-Wunused-function"
 #pragma clang diagnostic ignored "-Wunused-local-typedef"
+#pragma clang diagnostic ignored "-Wzero-length-array"
 #define FMT_HEADER_ONLY
 #include "distributed/mesh_device.hpp"
 #include "host_api.hpp"

